app/controllers/api/slots_controller.rb

```diff
@@ -1,8 +1,15 @@
 module Api
   class SlotsController < ApiController
     def index
-      prison = Prison.enabled.find(params[:prison_id])
-      @slots = prison.available_slots
+      prison = Prison.enabled.find(params.fetch(:prison_id))
+
+      slot_availability = SlotAvailability.new(prison: prison)
+      slot_availability.restrict_by_prisoner(
+        prisoner_number: params.fetch(:prisoner_number),
+        prisoner_dob: Date.parse(params.fetch(:prisoner_dob))
+      )
```

Comment:
Am I missing something?  I feel it would be cleaner if the prisoner info were passed into the initializer and the `slots` method automatically restricted the slots based on that.